### PR TITLE
adding displayWarningIcon ability to visual - api 3.7.0

### PR DESCRIPTION
**Pull request by anitamayorov**
Adding displayWarningIcon method which allows a visual to display a warning (info icon) to user. The warning is customized meaning the visual can send a title (hoverText) and details (detailedText) to be displayed. The limit of title is 70 characters and the limit of details is 2000 characters.

### CODE REVIEW

File: src/barChart.ts [hunk 1]
```diff
@@ -444,6 +460,12 @@ export class BarChart implements IVisual {
             .remove();
 
         this.handleClick(barSelectionMerged);
+
+        if (/\d/.test(settings.warningIconExample.text)) {
```
Comment (AviSander) [inline src/barChart.ts:464]: sorry - what is this syntax '/\d/' ?

**Review by AsafMozes — COMMENTED** [review]
Since this visual is supposed to represent a ready/useful visual, I suggest that instead of adding 
 warningIconExample, which does not serve any purpose besides checking the warning notification API, think of another way of utilizing the new functionality to indicate problem with the existing settings or data (like filter resulting 0 rows, or mismatching colors).
Otherwise add a new setting that would have some functionality and that can also be checked for validity (like e-mail or URL)

File: src/barChart.ts (hunk 1 above)
Comment (anitamayorov) [reply src/barChart.ts]: regex syntax - irrelevant after resolving comments

**Comment by anitamayorov** [comment]
@AsafMozes @AviSander 
added two use cases for the icon instead of the specific property pane category - one is the language and the other is the columns' colour

**Review by AsafMozes — APPROVED** [review]

**Review by AviSander — APPROVED** [review]

File: src/barChart.ts [hunk 2]
```diff
@@ -444,6 +444,21 @@ export class BarChart implements IVisual {
             .remove();
 
         this.handleClick(barSelectionMerged);
+
```
Comment (AviSander) [inline src/barChart.ts:447]: please add a comment to describe this

Comment (anitamayorov) [reply src/barChart.ts]: done

**Review by uve — APPROVED** [review]

**Pull request closed without merging**